hazelcast/src/main/java/com/hazelcast/map/AbstractEntryProcessor.java

```diff
@@ -65,10 +67,21 @@ public AbstractEntryProcessor(boolean applyOnBackup) {
         return entryBackupProcessor;
     }
 
-    private class EntryBackupProcessorImpl implements EntryBackupProcessor<K, V> {
+    private class EntryBackupProcessorImpl implements EntryBackupProcessor<K, V>, HazelcastInstanceAware {
```

Comment:
This will change the `serialVersionUID` of the EntryBackupProcessorImpl, so it won't deserialize in "emulated-mode" of a rolling upgrade, on a 3.8.x node.

What we could do is to have two implementations of the EntryBackupProcessorImpl, and check the clusterVersion in the EntryOperaiton, and do sth like this:
```
@Override
    public Operation getBackupOperation() {
       // ...
        EntryBackupProcessor backupProcessor = null;
        if(entryProcessor instanceOf AbstractEntryProcessor) {
                if(clusterVersion smaller than 3.9) {      
                         backupProcessor = entryProcessor.getBackupProcessor(); 
                } else {
                         // this class will be present on 3.9+ nodes
                         backupProcessor = new HazelcastAwareBackupEntryProcessorWrapper(entryProcessor);
                }
        }
        return backupProcessor != null ? new EntryBackupOperation(name, dataKey, backupProcessor) : null;
    }
```

What protects us is that the following AbstractEntryProcess's method is final:
```
public final EntryBackupProcessor<K, V> getBackupProcessor() {
        return entryBackupProcessor;
    }
```